runtime/src/staking.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2021 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): C. Yap
+
+use crate::Perbill;
+
+use codec::Decode;
+use frame_support::dispatch::{DispatchResultWithPostInfo, Vec};
+use frame_system::{Config as SysConfig, RawOrigin};
+use pallet_staking::{Pallet as StakingPallet, BalanceOf};
+use pallet_utxo::staking::StakingHelper;
+use sp_core::{H256, sp_std::vec};
+use sp_runtime::traits::StaticLookup;
+
+type StakeAccountId<T> =  <T as SysConfig>::AccountId;
+type LookupSourceOf<T> = <<T as SysConfig>::Lookup as StaticLookup>::Source;
+
+pub struct StakeOps<T>(sp_core::sp_std::marker::PhantomData<T>);
+impl <T: pallet_staking::Config + pallet_utxo::Config + pallet_session::Config> StakingHelper<T::AccountId> for StakeOps<T>
+    where StakeAccountId<T>: From<[u8; 32]>,
+          BalanceOf<T>: From<u128>
+{
+    fn get_account_id(pub_key: &H256) -> StakeAccountId<T> {
+        pub_key.0.into()
+    }
+
+    fn stake(stash_account: &StakeAccountId<T>, controller_account: &StakeAccountId<T>, session_key: &mut Vec<u8>, value: u128) -> DispatchResultWithPostInfo {
+        let controller_lookup: LookupSourceOf<T> = T::Lookup::unlookup(controller_account.clone());
+        let reward_destination = pallet_staking::RewardDestination::Staked;
+
+        // bond the funds
+        StakingPallet::<T>::bond(
+            RawOrigin::Signed(stash_account.clone()).into(),
+            controller_lookup,
+            value.into(),
+            reward_destination
```

Comment:
I kind of imagined the reward would go to the stash to be reused for staking. If one puts the stash as payee, will that reward be restaked?